src/Analyzer/ListNode.cpp

```diff
@@ -62,7 +63,21 @@ ASTPtr ListNode::toASTImpl(const ConvertToASTOptions & options) const
     expression_list_ast->children.resize(children_size);
 
     for (size_t i = 0; i < children_size; ++i)
-        expression_list_ast->children[i] = children[i]->toAST(options);
+    {
+        auto children_ast = children[i]->toAST(options);
+
+        // Named window should be wrapped in ASTWindowListElement with name.
+        if (children_ast->as<ASTWindowDefinition>())
+        {
+            auto window_list_node = std::make_shared<ASTWindowListElement>();
+            window_list_node->name = children[i]->getAlias();
+            window_list_node->children.push_back(children_ast);
+            window_list_node->definition = window_list_node->children.back();
+            children_ast = window_list_node;
+        }
```

Comment:
Am I correct that this may happen only for named windows? Let's then fix it in the `WindowNode::toAST`.
Let's wrap `ASTWindowDefinition` with `ASTWindowListElement` if `WindowNode` has an alias.

Reply:
I prefer removing the alias from WindowNode during query analysis. Thanks for the help with debugging.